executor/src/main/java/com/redhat/service/smartevents/executor/impl/CloudEventExecutorImpl.java

```diff
@@ -0,0 +1,109 @@
+
+import com.redhat.service.smartevents.executor.Executor;
+import com.redhat.service.smartevents.executor.MetricsConstants;
+import com.redhat.service.smartevents.executor.filters.FilterEvaluator;
+import com.redhat.service.smartevents.executor.filters.FilterEvaluatorFactory;
+import com.redhat.service.smartevents.infra.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.transformations.TransformationEvaluator;
+import com.redhat.service.smartevents.infra.transformations.TransformationEvaluatorFactory;
+import com.redhat.service.smartevents.infra.utils.CloudEventUtils;
+import com.redhat.service.smartevents.processor.actions.ActionInvoker;
+
+import io.cloudevents.CloudEvent;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+
+public class CloudEventExecutorImpl implements Executor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CloudEventExecutorImpl.class);
+
+    private final ProcessorDTO processor;
+    private final FilterEvaluator filterEvaluator;
+    private final TransformationEvaluator transformationEvaluator;
+    private final ActionInvoker actionInvoker;
+    private Timer processorProcessingTime;
+    private Timer filterTimer;
+    private Timer actionTimer;
+    private Timer transformationTimer;
+
+    public CloudEventExecutorImpl(ProcessorDTO processor, FilterEvaluatorFactory filterEvaluatorFactory, TransformationEvaluatorFactory transformationFactory,
+            ActionInvoker actionInvoker,
+            MeterRegistry registry) {
+        this.processor = processor;
+        this.filterEvaluator = filterEvaluatorFactory.build(processor.getDefinition().getFilters());
+        this.transformationEvaluator = transformationFactory.build(processor.getDefinition().getTransformationTemplate());
+        this.actionInvoker = actionInvoker;
+        initMetricFields(processor, registry);
+    }
+
+    @Override
+    public void onEvent(String event) {
+        onCloudEvent(CloudEventUtils.decode(event));
+    }
+
+    void onCloudEvent(CloudEvent cloudEvent) {
+        processorProcessingTime.record(() -> process(cloudEvent));
+    }
+
+    @SuppressWarnings("unchecked")
+    private void process(CloudEvent cloudEvent) {
+        LOG.info("Received event with id '{}' for Processor with name '{}' on Bridge '{}", cloudEvent.getId(), processor.getName(), processor.getBridgeId());
+
+        Map<String, Object> cloudEventData = CloudEventUtils.getMapper().convertValue(cloudEvent, Map.class);
+
+        // Filter evaluation
+        if (Boolean.TRUE.equals(filterTimer.record(() -> filterEvaluator.evaluateFilters(cloudEventData)))) {
+            LOG.info("Filters of processor '{}' matched for event with id '{}'", processor.getId(), cloudEvent.getId());
+
+            // Transformation
+            String eventToSend = transformationTimer.record(() -> transformationEvaluator.render(cloudEventData));
```

Comment:
@kostola For event sources, any transformation must produce another `CloudEvent`. Otherwise we potentially have events on the internal Kafka topic not in CloudEvent format.

Transformation/Filtering of Events from Sources are currently out-of-scope for your work, so we can postpone